### DISCUSSION TITLE
A (performance) quirk with JsonSerializer

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#47] [General]
# A (performance) quirk with JsonSerializer

System.Text.Json.JsonSerializer has a weird quirk in regard to performance and memory management. So we will discuss what is "wrong" with this code: JsonSerializer.Serialize(myObject, new JsonSerializerOptions(...));.

https://steven-giesel.com/blogPost/0e7512b0-c854-4cce-8606-765fcaf1e69c

**Comment by kyle69** [comment]
I want to say, it's quite logical to re-use an instance of JsonSerializerOptions, isn't it? ^^

**Reply by linkdotnet** [reply]
Well kind of - sure. If you would just consider the cost of `new`, then it is pretty neglectable. And that is where the surprise comes in.

And there are many samples where people new up inside the `Serialize` or `Deserialize` call: https://grep.app/search?q=%2C%20new%20JsonSerializerOptions&filter[lang][0]=C%23

So one could see this blog post as a reminder to reuse the instance and also as a small deep-dive why it behaves like it does.
